content/PreparedStatement/PreparedStatement.facts.md

```diff
@@ -0,0 +1,64 @@
+### Facts for Prepared Statement lesson:
+
+A `prepared statement` is used to compile an SQL statement once and executes multiple times with the same or different parameters. Parameters in a prepared statement can be substituted with `placeholders` which is helpful against SQL injection attacks.
+
+- A `placeholder` is often indicated by a question mark `?` called `positional placeholder` or a colon followed by a variable name `:variable_name` called `named placeholder`.
+
+- SQL injection `(SQLI)` is a code injection technique that alters SQL commands and exposes hidden data. It bypasses authentication then accesses, modifies, and deletes data in a database.
+
+The execution of a prepared statement consists of using `prepare()` and `execute()` methods. 
+
+- The `PDO::prepare()` method prepares an SQL statement for execution and returns a `PDOStatement` object.
+
+- The `PDOStatement::execute()` method executes the prepared statement.
+
+- The `PDOStatement::fetchAll()` method fetches all the result set rows.
+
+The example code below shows how to prepare and execute an SQL statement using different types of placeholder in PHP.
+
+1. Using `positional placeholders`
+
+```php
+$data = array (
+    'student1' => ['Berry', 'Meisner', '2002-09-09', 'Male'],
+    'student2' => ['Vikki', 'Fernando', '2005-05-17', 'Female']
+);
+        
+$pstmt = $conn->prepare("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES (?,?,?,?)");
+foreach ($data as $row) {
+    $pstmt->execute($row);
+}
+```
+- `$conn->prepare("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES (?,?,?,?)")` prepares the SQL statement.
+
+- `$pstmt = $conn->prepare("INSERT INTO students (first_name, last_name, birth_date, gender) VALUES (?,?,?,?)");` assigns the returned value of the `prepare()` method which is the `PDOStatement` object to the `$pstmt` variable.
+
+- The `VALUES (?,?,?,?)` clause uses positional placeholders `?` to substitute the values to be inserted into the table to prevent SQL injections.
```

Comment:
I think we still need to explain further how positional placeholder works and how it relates to the parameters passed to `execute`. E.g. if I have something like `VALUES(?, ?, ?, ?)` and when I do `$stmt->execute(['Berry', 'Meisner', '2002-09-09', 'Male'])` we should tell the learners that `Berry` will go to the first `?`, then `Meisner` will go to the 2nd `?` and so on.

Let us include also what will happen if we pass say `3` parameters and the prepared statement expects more than 3 or less than 3.